### PR TITLE
VAF masking

### PR DESCRIPTION
**Pull request by dngoldberg**
This P/R introduces a way to mask the VAF calculation done in solver.comp_Q_vaf() by introducing a logical mask.

The mask (variable vaf_mask) can be applied by setting:  

- params.error_prop.qoi_apply_vaf_mask = True
- params.io.vaf_mask_data_file to a netcdf file

the netcdf file must have x and y coordinates and the mask variable should be zero where the VAF should not be calculated and greater than zero (e.g. 1) where VAF is to be calculated, and should have a variable name given by params.io.vaf_mask_field_name (default "vaf_mask")

### CODE REVIEW

File: fenics_ice/model.py [hunk 1]
```diff
@@ -162,8 +162,8 @@ def init_fields_from_data(self):
 
         if self.params.melt.use_melt_parameterisation:
        
-         melt_depth_therm_const: float = -999.0
-         melt_max_const: float = -999.0
+         #melt_depth_therm_const: float = -999.0
+         #melt_max_const: float = -999.0
```
Comment (jrmaddison) [inline fenics_ice/model.py:166]: Is this change intended?

**Review by jrmaddison — COMMENTED** [review]
Seems fine, better to indent with 4 spaces. I'd prefer setting `var_mask` to `None` rather than reading `params`, but I see this pattern is consistent with other code.

File: fenics_ice/model.py (hunk 1 above)
Comment (dngoldberg) [reply fenics_ice/model.py:166]: @jrmaddison yes it is -- i think i copied and pasted from config.py to remind myself of param names and forgot to delete them. 

these parameters are only referred to below using the self.params object -- and so are not (i don't believe) used.

Comment (jrmaddison) [reply fenics_ice/model.py:166]: OK, I guess these lines can be deleted?

Comment (dngoldberg) [reply fenics_ice/model.py:166]: will do
